src/runtime/content.cpp

```diff
@@ -350,6 +373,52 @@ bool TrContentRuntime::recvXRCommand(int timeout)
   }
 }
 
+bool TrContentRuntime::recvInspectorCommand(int timeout)
+{
+  if (TR_UNLIKELY(inspectorCommandChanReceiver == nullptr))
+    return false;
+
+  inspector_comm::TrInspectorCommandMessage message;
+  if (inspectorCommandChanReceiver->recvCommand(message, timeout))
+  {
+    // Handle incoming inspector command message
+    handleInspectorCommandMessage(message);
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
+
+void TrContentRuntime::handleInspectorCommandMessage(inspector_comm::TrInspectorCommandMessage &message)
+{
+  switch (message.getType())
+  {
+  case inspector_comm::TrInspectorCommandType::CdpResponse:
+  {
+    auto cdpResponse = inspector_comm::TrInspectorCommandBase::CreateFromMessage<inspector_comm::TrCdpResponse>(message);
+    DEBUG(LOG_TAG_INSPECTOR, "Content %d: Received CDP response for request %u", id, cdpResponse.requestId);
+
+    // Forward response back to the ContentDomainProxy via static callback
+    ContentDomainProxy::handleResponseFromContent(cdpResponse.requestId, cdpResponse.cdpResponseJson);
+    break;
+  }
+  case inspector_comm::TrInspectorCommandType::CdpEvent:
+  {
+    auto cdpEvent = inspector_comm::TrInspectorCommandBase::CreateFromMessage<inspector_comm::TrCdpEvent>(message);
+    DEBUG(LOG_TAG_INSPECTOR, "Content %d: Received CDP event: %s", id, cdpEvent.cdpEventJson.c_str());
+
+    // Forward CDP events to inspector clients via the static handler
+    TrInspector::handleEventFromContent(cdpEvent.contentId, cdpEvent.cdpEventJson);
```

Comment:
`handleEventFromContent` should not be a static method, you can find the `TrInspector` instance from the constellation.